main/webservices/api/test_api.php

```diff
@@ -2,116 +2,531 @@
+
+        // compare each saved value to the original
+        $user = UserManager::getManager()->find($userId);
+        foreach($ORIGINAL_VALUES as $k => $v) {
+            $kk = ucfirst($k);
+            if ($k !== 'password' and method_exists($user, "get$kk")) {
+                $this->assertSame($v, eval("return \$user->get$kk();"), $k);
+            }
+        }
+
+        // assert user extra field value was saved
+        $extraFieldValueModel = new ExtraFieldValue('user');
+        $extraFieldValue = $extraFieldValueModel->get_values_by_handler_and_field_variable($userId, $EXTRA_FIELD_NAME);
+        $this->assertNotFalse($extraFieldValue);
+        $this->assertSame($EXTRA_FIELD_VALUE, $extraFieldValue['value']);
+
+        // clean up
+        UserManager::delete_user($userId);
+    }
+
+    /**
+     * @throws Exception if it cannot add the test courses to the test model session
+     */
+    public function testCreateSessionFromModel()
+    {
+        $ACTION = 'create_session_from_model';
+
+        // create a promotion
+        $career = new Career();
+        $careerId = $career->save([ 'name' => 'test career'.time() ]);
+        $promotion = new Promotion();
+        $promotionId = $promotion->save([ 'career_id' => $careerId, 'name' => 'test promo'.time() ]);
+
+        // create a model session
+        $modelSessionId = SessionManager::create_session(
+            'Model session'.time(),
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            '2019-01-01 00:00', '2019-08-31 00:00',
+            null, null
+        );
+
+        // move the session to a non-standard URL
+        if (api_is_multiple_url_enabled()) {
+            $urlId = UrlManager::add('https://www.url.org/chamilo-lms/' . time(), 'Non-default URL', 1);
+            UrlManager::update_urls_rel_session([$modelSessionId], $urlId);
+        }
+
+        // create an extra field and set its value in the model session - the new session will be given a different value in this field
+        $extraFieldModel = new ExtraField('session');
+        $EXTRA_FIELD_NAME = 'extraField'.time();
+        $EXTRA_FIELD_VALUE_FOR_MODEL_SESSION = 'extra field value for model';
+        $EXTRA_FIELD_VALUE_FOR_NEW_SESSION = 'extra field value for new session';
+        $extraFieldId = $extraFieldModel->save([
+            'field_type' => ExtraField::FIELD_TYPE_TEXT,
+            'variable' => $EXTRA_FIELD_NAME,
+            'display_text' => $EXTRA_FIELD_NAME,
+            'visible_to_self' => 1,
+            'visible_to_others' => 1,
+            'changeable' => 1,
+            'filter' => 1,
+        ]);
+        SessionManager::update_session_extra_field_value($modelSessionId, $EXTRA_FIELD_NAME, $EXTRA_FIELD_VALUE_FOR_MODEL_SESSION);
+
+        // create a second extra field and set its value in the model session - the new session will inherit the same value in this field
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 137 characters](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)